src/plots/cartesian/layout_attributes.js

```diff
@@ -344,6 +344,16 @@ module.exports = {
         editType: 'ticks',
         description: 'Determines whether or not the tick labels are drawn.'
     },
+    ticklabelsautomargin: {
```

Comment:
This might be too verbose. Would `(x|y)axis.automargin` suffice? Or maybe `tickautomargin`? Or perhaps we're looking at this the wrong way, something like:

```js

var layout = {
  automargin: true || false,
  automarginmode: 'all' || 'ticks' || 'title' || 'xaxis.title' || 'legend' || ...
  // or any combination of these using '+',  
  // a 'flaglist' attribute like scatter 'mode'
}
```

might scale better.

@alexcjohnson @cldougl @chriddyp any thoughts on this?

Reply:
+1 for `automargin` / `automarginmode` or at least `(x|y)axis.automargin` - this is more clear if attributes other than ticks/tick labels contributes to this (now or in the future).